stl/inc/vector

```diff
@@ -2882,6 +2882,63 @@ namespace pmr {
     using vector = _STD vector<_Ty, polymorphic_allocator<_Ty>>;
 } // namespace pmr
 #endif // _HAS_CXX17
+
+#if _HAS_IF_CONSTEXPR
+// VARIABLE TEMPLATE _Is_vb_iterator
+template <class _Alloc>
+_INLINE_VAR constexpr bool _Is_vb_iterator<_Vb_const_iterator<_Alloc>> = true;
+
+template <class _Alloc>
+_INLINE_VAR constexpr bool _Is_vb_iterator<_Vb_iterator<_Alloc>> = true;
+
+template <class _FwdIt, class _Ty>
+_CONSTEXPR20 void _Fill_vbool(_FwdIt _First, _FwdIt _Last, const _Ty& _Val) {
+    // Set [_First, _Last) to _Val
+    if (_First == _Last) {
+        return;
+    }
+
+    _Vbase* _UFirst            = const_cast<_Vbase*>(_First._Myptr);
+    const _Vbase* const _ULast = _Last._Myptr;
+
+    const auto _FirstSourceMask = _Vbase(-1) << _First._Myoff;
+    const auto _LastSourceMask  = _Vbase(-1) >> (_VBITS - _Last._Myoff);
+    const auto _FirstDestMask   = _Vbase(-1) ^ _FirstSourceMask;

```

Comment:
I guess it would be easier to read as `static_cast<_Vbase>(~_FirstSourceMask)`

Reply:
Also, `_Vbase(-1)` is a functional-syntax C-semantics cast, which we conventionally avoid.

Reply:
Thanks, replaced with `static_cast<_Vbase>`